Yoast/wpseo-woocommerce
# Add primary term check to the schema taxonomy

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the primary category would not be reflected correctly in the product schema output when it was being used to set the manufacturer or brand.

## Relevant technical choices:

* Added tests for general schema output to be able to test the private functions that are changed.
* Changed the class variables from private to protected to be able to set them in the test double.
* Moved the `Frontend->canonical` to a protected function to be able to mock it.
* Added PHPCS exclusions for the test doubles.

## Test instructions

This PR can be tested by following these steps:

* Reproduce on trunk: see the issue.
* Switch to this branch and build it.
* Refresh the frontend source of your product again and check the output of the `brand` and `manufacturer` to see if they match the primary category.

Fixes https://github.com/Yoast/bugreports/issues/468
